readparameters.h

```diff
@@ -22,57 +22,169 @@
 
 #ifndef READPARAMETERS_H
 #define READPARAMETERS_H
+
+#include <boost/lexical_cast.hpp>
+#include <boost/program_options.hpp>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
 #include <limits>
 #include <mpi.h>
 #include <stdint.h>
 #include <string>
+#include <typeinfo>
 #include <vector>
 
 #include "common.h"
+#include "version.h"
+
+namespace PO = boost::program_options;
```

Comment:
Hmm, in principle, it is bad style to declare a namespace shorthand like this in a header file.
After all, now every c++ file that includes it will also have the PO name resolved to boost::program_options.

I see that this is unavoidable for a template implementation (unless we want to spell out boost::program_options in every point it is used, which is not pretty, either), but I'm worried that "PO" might be an identifier somebody accidentally uses at one point, and is then confused by the error message.

Reply:
Indeed. I will remove using namespaces in headers.